web_api.md

```diff
@@ -0,0 +1,318 @@
+   or `path_to_privkey` (path to a private key). All Processing Workers, which will be deployed, must be declared under
+   the `deploy_processors` property. In case `type` is `docker`, make sure that [Docker](https://www.docker.com/) is
+   installed in the target machine and the provided `username` has enough rights to execute Docker commands.
+
+Among three sections, only the `message_queue` is required. However, if `hosts` is present, `mongo_db` must be there
+as well. For more information, please check the [configuration file schema](web_api/config.schema.yml).
+
+### Processing Worker
+
+There is normally no need to start (or stop) a Processing Worker manually, since it can be managed by a Processing
+Server via a [configuration file](#processing-server). However, if it is necessary to do so, there are two ways to start
+a Processing Worker:
+
+```shell
+# 1. Use ocrd CLI bundled with OCR-D/core
+$ ocrd processing-worker <processor-name> --queue=<queue-address> --database=<database-address>
+
+# 2. Use processor name
+<processor-name> processing-worker --queue=<queue-address> --database=<database-address>
+```
+
+* `--queue`: a [Rabbit MQ connection string](https://www.rabbitmq.com/uri-spec.html) to a running instance.
+* `--database`: a [MongoDB connection string](https://www.mongodb.com/docs/manual/reference/connection-string/) to a
+  running instance.
+
+### Process Queue
+
+By using a queuing system, specifically [RabbitMQ](https://www.rabbitmq.com/), the reliability and flexibility of the
+whole system are greatly improved. In our implementation, manual acknowledgement mode is used. This means, when a
+Processing Worker finishes successfully, it sends a positive ACK signal to RabbitMQ. In case of failure, it tries again
+three times before sending a negative ACK signal. When a negative signal is received, RabbitMQ will re-queue the
+message. If there is not any ACK signal sent for any reason (e.g. consumer crash, power outage, network problem, etc.),
+RabbitMQ will automatically re-queue the message after timeout, which is 30 minutes by default. This behavior, however,
+can be [overridden](https://www.rabbitmq.com/consumers.html#acknowledgement-timeout) by setting another value for
+the `consumer_timeout` property in the [`rabbitmq.conf`](https://www.rabbitmq.com/configure.html#config-file) file.
+
+To avoid processing the same input twice (in case of re-queuing), a Processing Worker first checks
+the [`redeliver`](https://www.rabbitmq.com/confirms.html#automatic-requeueing) property to see if this message was
+re-queued. If yes, and the status of this process in the database is not `SUCCESS`, it will process the data described
+in the message again.
+
+When a Processing Server receives a request, it creates a message based on the request content, then push it to an
+appropriate queue. A processing queue always has the same name as its respective processors. For
```

Comment:
What exactly does `appropriate queue` mean? Shouldn't this paragraph explain that a workflow must be decoded into single processor calls in a certain order, and that these calls then get pushed one by one onto the queue?

Reply:
> Shouldn't this paragraph explain that a workflow must be decoded into single processor calls in a certain order, and that these calls then get pushed one by one onto the queue?

That's better. I'll re-write it in this way.

Reply:
Thanks – that's really helpful!